wfe/web-front-end.go

```diff
@@ -1131,11 +1138,22 @@ func (wfe *WebFrontEndImpl) Certificate(ctx context.Context, logEvent *requestEv
 		return
 	}
 
+	parsedCertificate, err := x509.ParseCertificate([]byte(cert.DER))
+	if err != nil {
+		logEvent.AddError("unable to parse certificate: %s", err)
+		wfe.sendError(response, logEvent, probs.Malformed("Unable to parse certificate"), err)
+		return
+	}
+
 	addCacheHeader(response, wfe.CertCacheDuration.Seconds())
 
 	// TODO Content negotiation
 	response.Header().Set("Content-Type", "application/pkix-cert")
-	response.Header().Add("Link", link(IssuerPath, "up"))
+	if err = wfe.addIssuerCertificateURLs(response, parsedCertificate); err != nil {
```

Comment:
I think this should probably be moved up directly under the block where the certificate is parsed. That way we do these possibly-erroring steps before setting the cache and Content-Type headers.